tools/dvrescue.xsd

```diff
@@ -44,36 +50,180 @@
     <xsd:sequence>
       <xsd:element name="frame" type="frameType" minOccurs="0" maxOccurs="unbounded"/>
     </xsd:sequence>
-    <xsd:attribute name="count" type="xsd:integer"/>
-    <xsd:attribute name="pts" type="xsd:string"/>
-    <xsd:attribute name="end_pts" type="xsd:string"/>
-    <xsd:attribute name="size" type="xsd:string"/>
-    <xsd:attribute name="video_rate" type="xsd:string"/>
-    <xsd:attribute name="chroma_subsampling" type="xsd:string"/>
-    <xsd:attribute name="scanType" type="xsd:string"/>
-    <xsd:attribute name="aspect_ratio" type="xsd:string"/>
-    <xsd:attribute name="audio_rate" type="xsd:integer"/>
-    <xsd:attribute name="channels" type="xsd:integer"/>
+    <xsd:attribute name="scan_type" type="xsd:string">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          String representing the interlacement pattern of the frame.
+        </xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="count" type="xsd:integer">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          Frame number within the current frames sequence.
+        </xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="pts" type="xsd:string">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          Presentation timestamp for start of sequence.
+        </xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="end_pts" type="xsd:string">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          Presentation timestamp for end of sequence.
+        </xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="size" type="xsd:string">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          Dimensions of frame represented as width x height in pixels.
+        </xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="video_rate" type="xsd:string">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          Frame rate in fractional notation.
+        </xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="chroma_subsampling" type="xsd:string">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          Chroma subsampling value represented in three part ratio notation.
+        </xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="aspect_ratio" type="xsd:string">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          Frame aspect ratio represented in fractional notation.
+        </xsd:documentation>
+      </xsd:annotation>
+    </xsd:attribute>
+    <xsd:attribute name="audio_rate" type="xsd:integer">
+      <xsd:annotation>
+        <xsd:documentation xml:lang="en">
+          Frame audio rate represented in full integer notation.
```

Comment:
audio rate ("48000", "44100", "32000") is as video rate ("25"video rate for PAL, "30000/1001" for NTSC), with integer if possible, else ratio. Just that for the moment nobody was crazy enough for doing non integer rate for audio ;-). for DV specifically there is no more updates so no audio non integer rate, but if generic it could fail in the future. In practice not a big deal, more FYI.

Reply:
In non-mediainfo-type settings sometimes it's written as 48k or 44.1, so I was trying to define the resulting @audio_rate with that in mind.

Reply:
i prefer the audio rate as the integer in hz rather than a decimal in integer